app/controllers/candidate_interface/adviser_sign_ups_controller.rb

```diff
@@ -0,0 +1,39 @@
+module CandidateInterface
+  class AdviserSignUpsController < CandidateInterfaceController
+    before_action :set_adviser_sign_up
+    before_action :render_404_unless_available
+
+    def new; end
+
+    def create
+      if @adviser_sign_up.save
+        flash[:success] = t('application_form.adviser_sign_up.flash.success')
+
+        redirect_to candidate_interface_application_form_path
+      else
+        track_validation_error(@adviser_sign_up)
+        render :new
+      end
+    end
+
+  private
+
+    def set_adviser_sign_up
+      @adviser_sign_up = Adviser::SignUp.new(application_form, adviser_sign_up_params)
+    end
+
+    def application_form
+      current_candidate.current_application
+    end
+
+    def adviser_sign_up_params
+      params
+        .fetch(:adviser_sign_up, {})
```

Comment:
Should this be `.require(:adviser_sign_up)`?

Reply:
I used `fetch` internationally here as it means we can perform the availability check in a `before_action`. As the `before_action :render_404_unless_available` expects the `@adviser_sign_up` to be assigned the only way to do this in a shared `before_action` is with `fetch` (as `require` would raise an error on the `new` action given there are no parameters).